runtime/src/append_vec.rs

```diff
@@ -117,6 +117,10 @@ pub struct AppendVec {
 
 impl Drop for AppendVec {
     fn drop(&mut self) {
+        if env::var("SOLANA_DISABLE_DEAD_APPEND_VEC_REMOVAL").is_ok() {
```

Comment:
Well, granted I'm fond of `env`s. ;)

@sakridge I guess this is neligible for the performance-wise.

Reply:
I would prefer a member variable here.

Reply:
Something like `new_empty_map(len: usize, cleanup: bool) {}` and `fn drop(self) { if self.cleanup { remove_file(x); }`

Reply:
@sakridge @mvines 

Well, that knob is [needed for `solana-validator`](https://github.com/solana-labs/solana/pull/9490/files#diff-657d1e7648a9fb74cb1582a9c48beb4bR121) not for `solana-ledger-tool`. So, I have to resort to an env var, otherwise, some considerable plumbing is needed or `lazy_static!`...

Btw, `solana-ledger-tool append-vec` wasn't oddly removing the specified `AppendVec`s. Then, I found a footgun:  [the `exit(0)` immediately after it](https://github.com/solana-labs/solana/pull/9490/commits/dc845e032a6ddbc932fc26df6fed9d4dd2e2467c#diff-3125b196bc2e46166465885cd9ad66adR954) caused the process to exit without calling `Drop::drop()`s altogether... So, it was working as intended by mistake.

A bit scared of the misstep, I've refactored and fixed that with tests, resulting rather bigger PR... :)

Also, I've chosen `read_only` (`false`, by default) instead of `cleanup` (`true`, by default) because I think the naming more directly conveys the behavior and I generally prefer that default behavior equates to `Default::default()` value for compatibility, less cognitive load, and readability.